### DISCUSSION TITLE
NanoEls H4 build with dividing attachement

### DISCUSSION THREAD

**Discussion by fmw626** [kachurovskiy/nanoels#123] [Show and tell]
Hello everyone,

I have upgraded my [H2 build](https://github.com/kachurovskiy/nanoels/discussions/118) with the NanoEls H4.

![IMG_7330](https://github.com/kachurovskiy/nanoels/assets/122809499/664106f9-aff6-485f-9426-68cc42fea237)

First impressions of the H4 are overwhelmingly positive. It surpasses the capabilities of the H2 by a wide margin, making it an absolute joy to use. One standout feature is the exceptional backlash compensation, although I have only been using it for a day, I can already confidently say that I wouldn't want to go back to a device without this feature.

Here are the parts that I made for the upgrade (don't mind the black and white color scheme, I ran out of black filament):
- [X axis adjustable nema17 motor mount (belt drive)](https://www.thingiverse.com/thing:6058899)
      Initially printed in PLA, I encountered some softening due to the heat generated by the stepper motor. To address this, I am currently printing a PETG part, which I hope will solve the issue. Additionally, I found that the original two-piece metal handle caused excessive rattling during stepper motor operation, so I removed it. I am considering turning a spring to prevent the handle from rattling. There is a slight interference between the handle from the manual Z-axis wheel and the mount. I have temporarily removed the handle and plan to slightly modify the wheel to prevent future collisions.
- [Z axis adjustable nema24 motor bracket (belt drive)](https://www.thingiverse.com/thing:6075086)
- [NanoEls H4 Seat](https://www.thingiverse.com/thing:6075055)
- Work in progress dividing attachment

I will write [another post](https://github.com/kachurovskiy/nanoels/discussions/124) in Q&A regarding my feedback & questions about the H4.

fmw

![IMG_7331](https://github.com/kachurovskiy/nanoels/assets/122809499/3938b8aa-c286-48d7-b799-d5608f838a5b)
![IMG_7334](https://github.com/kachurovskiy/nanoels/assets/122809499/03ddf022-1562-44d4-ad46-d5c0aa6366a5)
![IMG_7343](https://github.com/kachurovskiy/nanoels/assets/122809499/7b5d4878-5341-49ae-baac-032403fe3e64)
![IMG_7335](https://github.com/kachurovskiy/nanoels/assets/122809499/d09c2ec1-5a70-41fd-9e9d-e15a732f8a6b)
![IMG_7316](https://github.com/kachurovskiy/nanoels/assets/122809499/0b52992a-e116-4e77-861e-206d05b339b4)
![IMG_7315](https://github.com/kachurovskiy/nanoels/assets/122809499/27785680-4fed-40ad-9354-33503f24e29b)

**Comment by kachurovskiy** [comment]
Hi @fmw626, what an awesome build, thank you so much for sharing. Special thanks for all the amazing models on Thingiverse 🤩 So cool to see the actual stock being processed.

PLA does soften in contact with motor during continuous use, I saw that too.

Rattling issues were solved by slipping small black o-rings on all the handles:

![image](https://github.com/kachurovskiy/nanoels/assets/517919/82ae8763-9d55-4797-b9c9-d47d49d3fac9)

**Reply by fmw626** [reply]
> Special thanks for all the amazing models on Thingiverse 🤩 

Of course! More to come....


> So cool to see the actual stock being processed.

Parting is going so much smoother now than by hand 🥳 👍 


> Rattling issues were solved by slipping small black o-rings on all the handles:

Genius. I'll give this a try!

**Comment by fmw626** [comment]
Since @kachurovskiy implemented the dividing head mode into the h4 code, i had the chance to play around with it today. It works 
great, so here I am sharing the dividing head design: https://www.thingiverse.com/thing:6086206


The build is a bit complex with many parts and probably needs some adjustments for your specific lathe. Thus, I only provide the fusion 360 .f3d file without timeline (it is a mess of revisions) for now. Be prepared to edit the spacer lengths and the housing hole pattern to your lathe. Bear in mind that my lathe was already modified before I added the dividing head (new spindle pulleys, no more hall sensor on the spindle, and more…). The pulley cover won’t fit on the lathe anymore without extensive modifications.


It works as follows:
- A 180t worm wheel is attached to the back of the spindle. I previously turned oversized spindle drive pulleys, so it was just a matter of adding some threaded holes to the pulley. Be sure to have the worm wheel be concentric with your spindle!
- A short 10mm shaft is secured through the thick metal plate on the left side of the lathe. I believe the hole I used was initially to route cables into the electronic box. 
- The longer 10mm worm shaft assembly pivots on the previously mentioned shaft, allowing to engage/disengage the worm gear.
- the pivoting movement of the shaft is limited by a custom housing. The worm shaft can be secured in the engaged/disengaged position through a wedge
- Software Integration with NanoELS H4 is in progress. Thank you, @kachurovskiy. 

The part "Shaft base" holds the stepper, and should be printed in heat-resistant filament. The rest should be fine in PLA.

Additional parts I used for my build (might vary for yours!) 

- Turned brass worm gear: major diameter = 25mm, minor diameter = 21,85mm, pitch= 2,2mm, inner diameter = 10mm (+2 m3 threaded holes for grub screws)
- 1x ≈280mm 10mm steel rod, with m10 threads at both ends (+ keyway/flat cut/files in for grub screws)
- 1x ≈135mm 10mm steel rod, with m10 threads at both ends

- 1x nema 17 stepper (I used 60mm)
- HtdHTD 3m 56t (168mm) belt (9mm wide)
- HTD 3M-20BF-B5-10: 20t, Typ BF, ø 5mm pulley 
- HTD 3M-30BF-B10-10: 30t, Typ BF, ø 10mm pulley
- 6x M10 nuts
- 11x M10 din 440r washers (≈11x34x2,6mm)
- 1x M10 din 9021 washers (≈11x29,5x2,5mm)
- 4x 6000-2RS Ball Bearings Double Sealed 10 mm x 26 mm x 8 mm
- 2x F10-18M Axial Ball Bearings 10 mm x 18 mm x 5.5 mm
- Various m3 hardware (cylinder head screws, grub screws, washers, nuts, brass inserts)
- 1x m4 threaded rod (about 25/30mm) + washers & nuts (secures the housing at the front, into the threaded hole where the hall sensor used to be on the stock lathe)
- 2x m5 x 100mm cylinder head screws + washers + nuts (holds the dividing head housing together)
- 5x 1,5mm x 6mm steel dowel pins (optional, to align the dividing head halves perfectly. Use a 1,5mm hand drill to ream the holes in the printed parts for a good fit)
- 1 piece of m10 threaded rod (25-30mm). Additional fastener to secure the dividing head housing to the lathe, through a former pulley-cover-bolt-hole
- Printed adhesive scales for 0-360° (worm wheel) and 0-2° (worm shaft)
- Piece of wire as an indicator for the 0-360° scale

![Screenshot 2023-06-19 at 18 18 23](https://github.com/kachurovskiy/nanoels/assets/122809499/97338d4b-d74e-45ea-85be-f2f37fbdf19f)
![Screenshot 2023-06-19 at 18 18 07](https://github.com/kachurovskiy/nanoels/assets/122809499/b1b2740f-a2b8-4fa1-99a5-09e3d946852b)
![Screenshot 2023-06-19 at 18 07 48](https://github.com/kachurovskiy/nanoels/assets/122809499/c1029b3a-8c22-4e31-a7dd-0cdf956f0c63)
![Screenshot 2023-06-19 at 17 43 15](https://github.com/kachurovskiy/nanoels/assets/122809499/19281e66-afce-4db6-b9a4-3c523cbb43c9)

**Reply by kachurovskiy** [reply]
That's a massive build, very impressive, thanks a lot for sharing 👍

I didn't push the dividing head code update to the main repo yet, needs a bit more work. Coming soon 😀

**Reply by fmw626** [reply]
Here is a C-axis button label sign thingy to put above the H4

![Screenshot 2023-06-20 at 01 32 14](https://github.com/kachurovskiy/nanoels/assets/122809499/07dcde5a-70c8-41b5-8662-8a839e0dcac6)

[C_Button_Labels.stl.zip](https://github.com/kachurovskiy/nanoels/files/11793945/C_Button_Labels.stl.zip)

One thing that is a bit confusing is that with the C axis we move the workpiece and not the tool, thus the movements can be counterintuitive. I have mapped the axis such that the rotation of the spindle corresponds to the icons, but the limit buttons are currently mapped the wrong way around, so that the UP LIMIT is reached when pressing DOWN and vice versa.

**Reply by kachurovskiy** [reply]
Thank you @fmw626, it's beautiful 👍

Yeah, I aimed at arrows to make sense physically and match with H4 notion of angle incrementing when spindle is moving forward with the choice of arrow direction on

![image](https://github.com/kachurovskiy/nanoels/assets/517919/45a45a6e-2e54-45cf-93f3-0de062d1e2e1)
